### ISSUE TITLE
Unloading a solution while it's being loaded by the roslyn workspace can lead to NRE

### ISSUE THREAD

**Issue by Therzok** [mono/monodevelop#7211]
Dispose on the MonoDevelopWorkspace nulls out the solution, so pass it through the load handler. This keeps the behaviour of not retaining the solution object.

```
ERROR [2019-02-21 16:30:27Z]: Could not load parser database.
  System.AggregateException: One or more errors occurred. (Value cannot be null.
  Parameter name: key) ---> System.ArgumentNullException: Value cannot be null.
  Parameter name: key
    at System.Collections.Concurrent.ConcurrentDictionary`2[TKey,TValue].ThrowKeyNullException () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs:918 
    at System.Collections.Concurrent.ConcurrentDictionary`2[TKey,TValue].TryGetValue (TKey key, TValue& value) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs:441 
    at MonoDevelop.Ide.TypeSystem.TypeSystemService.InternalLoad (System.Collections.Generic.List`1[T] mdWorkspaces, MonoDevelop.Core.ProgressMonitor progressMonitor, System.Threading.CancellationToken cancellationToken, System.Boolean showStatusIcon) [0x000cb] in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService_WorkspaceHandling.cs:155 
    at MonoDevelop.Ide.TypeSystem.TypeSystemService.Load (MonoDevelop.Projects.WorkspaceItem item, MonoDevelop.Core.ProgressMonitor progressMonitor, System.Threading.CancellationToken cancellationToken, System.Boolean showStatusIcon) [0x0006a] in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService_WorkspaceHandling.cs:122 
     --- End of inner exception stack trace ---
  ---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
  Parameter name: key
    at System.Collections.Concurrent.ConcurrentDictionary`2[TKey,TValue].ThrowKeyNullException () [0x00000] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs:918 
    at System.Collections.Concurrent.ConcurrentDictionary`2[TKey,TValue].TryGetValue (TKey key, TValue& value) [0x00008] in /Users/builder/jenkins/workspace/build-package-osx-mono/2018-08/external/bockbuild/builds/mono-x64/external/corefx/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs:441 
    at MonoDevelop.Ide.TypeSystem.TypeSystemService.InternalLoad (System.Collections.Generic.List`1[T] mdWorkspaces, MonoDevelop.Core.ProgressMonitor progressMonitor, System.Threading.CancellationToken cancellationToken, System.Boolean showStatusIcon) [0x000cb] in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService_WorkspaceHandling.cs:155 
    at MonoDevelop.Ide.TypeSystem.TypeSystemService.Load (MonoDevelop.Projects.WorkspaceItem item, MonoDevelop.Core.ProgressMonitor progressMonitor, System.Threading.CancellationToken cancellationToken, System.Boolean showStatusIcon) [0x0006a] in /Users/alan/Projects/monodevelop/main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.TypeSystem/TypeSystemService_WorkspaceHandling.cs:122 <---
```

> VS bug [#797986](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/797986)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-6] Comment by **xamarinc**



Fixed in version 8.0.0.1639 (master)



Author: Marius Ungureanu  
 Commit: 557371b4febbf836bac5a877d744daa6b87788ff (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-8] Comment by **xamarinc**



Fixed in version 8.0.0.2453 (release-8.0)



Author: Marius Ungureanu  
 Commit: 557371b4febbf836bac5a877d744daa6b87788ff (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-9] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Author: Marius Ungureanu  
 Commit: 557371b4febbf836bac5a877d744daa6b87788ff (xamarin/monodevelop)
